index.js

```diff
@@ -750,6 +788,174 @@ class GoogleAuth extends DbBase {
     const admin = await this._getAdmin(email)
     return !!admin?.password
   }
+
+  // TODO: move daily limits logic into its own module and invoke it it here?
+
+  /**
+   * Creates or update a daily limit configuration for a given combination of admin level and daily limit category
+   * @param {number} level - The admin level
+   * @param {DailyLimitCategory} category - The daily limit category
+   * @param {DailyLimitConfig} config - The configuration values for the admin level and category daily limit
+   * @throws {UserError} In the following cases:
+   * - admin level is not integer or it's not between 0 and 4 inclusive
+   * - category is neither `opened` nor `displayed`
+   * - `config.alert` and `config.block` are not provided, or at least one of them is provided but not integer or it's integer but not positive
+   * - the daily limit for the given admin level and category exists but `config.alert` and `config.block` are not provided
+   * @returns {Promise<boolean>} Resolves to `true` if daily limit configuration is created/updated successfully. Otherwise, triggers a rejection.
+   */
+  async setAdminLevelDailyLimit (level, category, config) {
+    this._validateAdminLevel(level)
+    this._validateDailyLimitCategory(category)
+
+    const existingLevelDailyLimit = await new Promise((resolve, reject) => {
```

Comment:
instead of duplicating the db get, can do
```suggestion
    const existingLevelDailyLimit = await this.getAdminLevelDailyLimit(level, category)
```